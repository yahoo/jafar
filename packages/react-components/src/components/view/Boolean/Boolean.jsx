/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import React from 'react';
import PropTypes from 'prop-types';
import TrueIcon from '@material-ui/icons/CheckCircleOutline';
import FalseIcon from '@material-ui/icons/NotInterested';
import Styled from './StyledComponents';

/**
 * Represent a boolean value
 * 
 * Import <a target="_blank" 
 href="https://github.com/yahoo/jafar/blob/master/packages/react-components/src/components/view/Boolean/Boolean.jsx">
 Boolean</a> from '@jafar/react-components/view/Boolean'
 */
export default class Boolean extends React.Component {
  static propTypes = {
    value: PropTypes.bool,
  };

  static defaultProps = {
    value: false,
  };

  render() {
    const Icon = this.props.value ? TrueIcon : FalseIcon;
    const color = this.props.value ? '#1adc91' : '#989697';

    return (<Styled.Icon color={color}><Icon /></Styled.Icon>);
  }
}
