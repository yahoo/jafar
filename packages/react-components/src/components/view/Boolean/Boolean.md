<h3>Examples</h3>

<h4>Usage in jafar form</h4>

```javascript
const Form = require('@jafar-org/react-form/Form').default;
const Field = require('@jafar-org/react-form/Field').default;

const model = {
  id: 'simple',
  fields: {
    canEdit: {
      label: 'Can Edit',
      path: 'canEdit',
      component: {
        name: 'Boolean',
      }
    },
  },
  data: {
      canEdit: true,
  },
};

const resources = {
  components: { 
    Boolean: {
        renderer: Boolean,
    }, 
  }
};

<Form model={model} resources={resources}>
    <Field id='canEdit' />
</Form>
```

<h4>Value: true</h4>

```javascript
initialState = { 
    value: true
};

<Boolean value={state.value} />
```

<h4>Value: false</h4>

```javascript
initialState = { 
    value: false
};

<Boolean value={state.value} />
```

<h4>Value: undefined</h4>

```javascript
initialState = { 
    value: undefined
};

<Boolean value={state.value} />
```
