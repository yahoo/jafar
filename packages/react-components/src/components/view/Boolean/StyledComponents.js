/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import styled from 'styled-components';

export const Icon = styled.div`
  color: ${props => props.color};
`;

export default {
  Icon,
};
