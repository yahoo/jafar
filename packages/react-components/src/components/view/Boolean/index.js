/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import Boolean from './Boolean';

export default Boolean;
