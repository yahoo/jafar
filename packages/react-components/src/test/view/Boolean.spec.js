/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import React from 'react';
import { shallow } from 'enzyme';
import Boolean from '../../components/view/Boolean/index';

describe('<Boolean />', () => {
  let component;
  const value = true;

  beforeEach(() => {
    component = shallow(
      getComponent(value)
    );
  });
  it('should render provided data', () => {
    expect(component).toMatchSnapshot();
  });

  function getComponent(value) {
    return (
      <Boolean value={value} />
    );
  }
});
