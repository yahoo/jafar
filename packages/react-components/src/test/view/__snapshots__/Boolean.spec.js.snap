// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Boolean /> should render provided data 1`] = `
<styled.div
  color="#1adc91"
>
  <CheckCircleOutlineIcon />
</styled.div>
`;
